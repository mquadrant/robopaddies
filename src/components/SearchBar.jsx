import React from 'react'

export default function SearchBar(props) {
    const {searchChange}=props;
    return (
        <div>
            <input className="shadow appearance-none border rounded w-36 py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline" type='text' name='search' placeholder='Search Robots' onChange={searchChange}/>
        </div>
    )
}
