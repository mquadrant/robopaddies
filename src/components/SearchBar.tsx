import React from 'react'

interface ISearchBoxProps {
    searchChange(event:React.SyntheticEvent<HTMLInputElement>):void
}

export default function SearchBar(props:ISearchBoxProps) {
    const {searchChange}:ISearchBoxProps=props;
    return (
        <div>
            <input className="shadow appearance-none border rounded w-36 py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline" type='text' name='search' placeholder='Search Robots' onChange={searchChange}/>
        </div>
    )
}
