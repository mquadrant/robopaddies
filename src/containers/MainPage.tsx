import React ,{useState,useEffect}from 'react';
import CardList from '../components/CardList'
import SearchBar from '../components/SearchBar'
import Scroll from '../components/Scroll'
import ErrorBoundry from '../components/ErrorBoundry';



export default function MainPage(props:any) {
  const {requestForRobots,robots} = props;
  const [filteredRobots, setFilteredRobots] = useState([]);
  
  useEffect(()=>{ 
    requestForRobots();
  },[requestForRobots])

  useEffect(()=>{ 
    setFilteredRobots(robots);
  },[robots])

  const searchHandler = (e:React.ChangeEvent<HTMLInputElement>):void=>{ 
    filterRobots(e.target.value);
  }

  const filterRobots = (term:string)=>{
    let re = new RegExp(term, "gi");
    setFilteredRobots(robots.filter((robot:any)=>{
      return re.test(robot.name) || re.test(robot.email)
    }))
  }
  
  return (
          <div className="app-content">
            <div className="text-center my-12">
            <h1 className="text-5xl my-8">ROBOPADDIES</h1>
            <SearchBar searchChange={searchHandler}/>
            </div>
          <hr className="border-green-200"/>
          {filteredRobots.length?(
          <Scroll>
            <ErrorBoundry>
              <CardList robots={filteredRobots}/>
            </ErrorBoundry>
          </Scroll>):null}
        </div>
  )
    
}
