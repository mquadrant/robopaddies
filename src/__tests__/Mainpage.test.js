import React from 'react';
import { shallow,mount} from 'enzyme';
import MainPage from '../containers/MainPage';
import SearchBar from '../components/SearchBar'

let wrapper;

beforeEach(()=>{
    const mockProps = {
        requestForRobots: jest.fn(),
        robots:[]
    }
    wrapper = shallow(<MainPage {...mockProps}/>);
})

it('renders without crashing', () => {
  expect(wrapper).toMatchSnapshot();
});

it('renders one <SearchBar /> components', () => {
    const wrapper = shallow(<MainPage />);
    expect(wrapper.find(SearchBar)).toHaveLength(1);
  });
  
it('run inner function components', () => {
  const prop = {
    requestForRobots:jest.fn().mockResolvedValue(),
    robots:[{name:"john",email:"jemail@hmail.com"}]
  }
    const wrapper = shallow(<MainPage {...prop}/>);

    expect(wrapper.find(SearchBar)).toHaveLength(1);
    expect(wrapper.find(SearchBar).prop('searchChange')({
          target: {
            value: 'e',
          },
        })).toBeUndefined();
  });
