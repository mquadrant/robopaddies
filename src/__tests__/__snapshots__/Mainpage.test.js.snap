// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="app-content"
>
  <div
    className="text-center my-12"
  >
    <h1
      className="text-5xl my-8"
    >
      ROBOPADDIES
    </h1>
    <SearchBar
      searchChange={[Function]}
    />
  </div>
  <hr
    className="border-green-200"
  />
</div>
`;
